src/praisonai-agents/tests/test_architectural_fixes.py

```
@@ -0,0 +1,440 @@
+        sequential_results = sequential_executor.execute_batch(tool_calls, mock_execute_tool_fn)
+        sequential_duration = time.time() - start_time
+        
+        # Test parallel execution
+        parallel_executor = ParallelToolCallExecutor(max_workers=3)
+        start_time = time.time()
+        parallel_results = parallel_executor.execute_batch(tool_calls, mock_execute_tool_fn)
+        parallel_duration = time.time() - start_time
+        
+        # Verify results are the same
+        assert len(sequential_results) == len(parallel_results) == 3
+        for seq_result, par_result in zip(sequential_results, parallel_results):
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)